--- a/src/capnpc-zig/struct_gen.zig
+++ b/src/capnpc-zig/struct_gen.zig
@@ -1565,7 +1565,7 @@
             try writer.print("{s}}}\n\n", .{member_indent});
         }
         try writer.print("{s}}};\n\n", .{decl_indent});
-        try writer.print("{s}pub fn brands(self: @This()) Brands {{\n", .{decl_indent});
+        try writer.print("{s}pub fn brands(self: @This()) @This().Brands {{\n", .{decl_indent});
         try writer.print("{s}return .{{ ._reader = self._reader }};\n", .{member_indent});
         try writer.print("{s}}}\n\n", .{decl_indent});
     }
@@ -1658,7 +1658,7 @@
             try writer.print("{s}}}\n\n", .{member_indent});
         }
         try writer.print("{s}}};\n\n", .{decl_indent});
-        try writer.print("{s}pub fn brands(self: @This()) Brands {{\n", .{decl_indent});
+        try writer.print("{s}pub fn brands(self: @This()) @This().Brands {{\n", .{decl_indent});
         try writer.print("{s}return .{{ ._builder = self._builder }};\n", .{member_indent});
         try writer.print("{s}}}\n\n", .{decl_indent});
     }
